server/src/main/java/org/elasticsearch/index/codec/vectors/diskbbq/IVFVectorsWriter.java

```diff
@@ -53,12 +53,18 @@ public abstract class IVFVectorsWriter extends KnnVectorsWriter {
     private final IndexOutput ivfCentroids, ivfClusters;
     private final IndexOutput ivfMeta;
     private final String rawVectorFormatName;
+    private final boolean useDirectIOReads;
     private final FlatVectorsWriter rawVectorDelegate;
 
     @SuppressWarnings("this-escape")
-    protected IVFVectorsWriter(SegmentWriteState state, String rawVectorFormatName, FlatVectorsWriter rawVectorDelegate)
-        throws IOException {
+    protected IVFVectorsWriter(
```

Comment:
we need bwc tests here for format versions from before `VERSION_DIRECT_IO`

So, might I suggest having version settable (package private) for the writer that way we can have tests that purposefully write an index with this old version, then force the reader to read to make sure we are bwc.